bin/bat_taxo_process.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+import os
+import argparse
+
+
+##### This script generates the final output of taxonomy annotation for eukaryotic MAGs from BAT outputs
+##### Alejandra Escobar, EMBL-EBI
+##### July 4, 2023
+
+def taxo_reporter( out_file, bat_files ):
+    with open(out_file, 'w') as to_print, open('all_bin2classification.txt', 'w') as to_concat:
+        to_print.write("\t".join(
+            [
+                'bin',
+                'lineage_taxids',
+                'lineage_names'
+            ])+'\n')
+        to_concat.write("\t".join(
+            [
+                '# bin',
+                'classification',
+                'reason',
+                'lineage',
+                'lineage scores'
+            ])+'\n')
+        for current_bin in bat_files:
+            with open(current_bin, 'r') as file_in:
+                next(file_in)
+                for line in file_in:
+                    line = line.rstrip().split("\t")
+                    concat_list = []
+                    genome = line.pop(0)
+                    concat_list.append(genome)
+                    classification = line.pop(0)
+                    concat_list.append(classification)
+                    reason = line.pop(0)
+                    concat_list.append(reason)
+                    lineage = line.pop(0)
+                    concat_list.append(lineage)
+                    lineage_scores = line.pop(0)
+                    concat_list.append(lineage_scores)
+                    to_concat.write("\t".join(concat_list)+'\n')
```

Comment:
To be honest, I don't understand why do you use pop(0) five times. Did you need just first 5 values from `line`
would `concat_list = line[:5]` work?

Reply:
Thank you for your comment, the code looks so much cleaner now.